plugins/azure_ad_admin/icon_azure_ad_admin/triggers/risk_detection/trigger.py

```diff
@@ -17,67 +26,123 @@ def __init__(self):
             input=RiskDetectionInput(),
             output=RiskDetectionOutput(),
         )
-        self.risk_level = ""
-        self.found = {}
 
-    def initialize(self):
-        new_risks = self.get_risks()
+    def run(self, params={}):
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        risk_level = params.get(Input.RISK_LEVEL)
+        frequency = params.get(Input.FREQUENCY, 60)
+        # END INPUT BINDING - DO NOT REMOVE
 
-        try:
-            result = new_risks["value"]
-        except KeyError:
-            raise PluginException(
-                cause="Unexpected output format.",
-                assistance="The output from Azure Active Directory was not in the expected format. Please contact support for help.",
-                data=new_risks,
+        # Initialize the trigger with starting data point
+        current_risks = self._get_risks(risk_level)
+        latest_risk_detection_time = self._get_latest_risk_detection_time(current_risks)
+
+        while True:
+            # Pull latest riskDetection data
+            self.logger.info("Pulling risks data from the API...")
+            new_risks = self._get_risks(risk_level, latest_risk_detection_time)
+
+            # Filter all the results by their detectedDateTime, only results that their time greater than
+            # previous latest risk detection detectedDateTime are returned
+            compared_risks = list(
```

Comment:
Should we add logging for un-compared/compared risks?

Reply:
What do you mean? 😄